src/main/java/net/minecraftforge/fluids/FluidAttributes.java

```diff
@@ -200,27 +269,29 @@ public boolean doesVaporize(BlockAndTintGetter reader, BlockPos pos, FluidStack
     }
 
     /**
-     * Called instead of placing the fluid block if {@link net.minecraft.world.dimension.Dimension#doesWaterVaporize()} and {@link #doesVaporize(FluidStack)} are true.
+     * Called instead of placing the fluid block if {@link net.minecraft.world.level.dimension.DimensionType#ultrawarm()} and {@link #doesVaporize(BlockAndTintGetter, BlockPos, FluidStack)} are true.
      * Override this to make your explosive liquid blow up instead of the default smoke, etc.
-     * Based on {@link net.minecraft.item.BucketItem#tryPlaceContainedLiquid(PlayerEntity, World, BlockPos)}
+     * Based on {@link net.minecraft.world.item.BucketItem#emptyContents(Player, Level, BlockPos, BlockHitResult)}
      *
      * @param player     Player who tried to place the fluid. May be null for blocks like dispensers.
-     * @param worldIn    World to vaporize the fluid in.
-     * @param pos        The position in the world the fluid block was going to be placed.
+     * @param level      Level to vaporize the fluid in.
+     * @param pos        The position in the level the fluid block was going to be placed.
      * @param fluidStack The fluidStack that was going to be placed.
      */
-    public void vaporize(@Nullable Player player, Level worldIn, BlockPos pos, FluidStack fluidStack)
+    public void vaporize(@Nullable Player player, Level level, BlockPos pos, FluidStack fluidStack)
     {
-        worldIn.playSound(player, pos, SoundEvents.FIRE_EXTINGUISH, SoundSource.BLOCKS, 0.5F, 2.6F + (worldIn.random.nextFloat() - worldIn.random.nextFloat()) * 0.8F);
+        level.playSound(player, pos, SoundEvents.FIRE_EXTINGUISH, SoundSource.BLOCKS, 0.5F, 2.6F + (level.random.nextFloat() - level.random.nextFloat()) * 0.8F);
 
         for (int l = 0; l < 8; ++l)
         {
-            worldIn.addAlwaysVisibleParticle(ParticleTypes.LARGE_SMOKE, (double) pos.getX() + Math.random(), (double) pos.getY() + Math.random(), (double) pos.getZ() + Math.random(), 0.0D, 0.0D, 0.0D);
+            level.addAlwaysVisibleParticle(ParticleTypes.LARGE_SMOKE, (double) pos.getX() + Math.random(), (double) pos.getY() + Math.random(), (double) pos.getZ() + Math.random(), 0.0D, 0.0D, 0.0D);
         }
     }
 
     /**
-     * Returns the localized name of this fluid.
+     * Returns the localized name of the provided {@code FluidStack}.
+     * @param stack the provided {@code FluidStack} to get the display name of.
+     * @return Returns the display name of the provided {@code FluidStack} as a {@code Component}.
      */
     public Component getDisplayName(FluidStack stack)
```

Comment:
The body of this method (while not touched by this PR) should be fixed to pass the stack to the `getTranslationKey` method being used.